awx_collection/plugins/lookup/tower_api.py

```diff
@@ -0,0 +1,195 @@
+__metaclass__ = type
+
+DOCUMENTATION = """
+lookup: tower_api
+author: John Westcott IV (@john-westcott-iv)
+short_description: Search the API for objects
+requirements:
+  - None
+description:
+  - Returns GET requests from the Ansible Tower API. See
+    U(https://docs.ansible.com/ansible-tower/latest/html/towerapi/index.html) for API usage.
+  - For use that is cross-compatible between the awx.awx and ansible.tower collection
+    see the tower_meta module
+extends_documentation_fragment: awx.awx.auth_plugin
+options:
+  _terms:
+    description:
+      - The endpoint to query, i.e. teams, users, tokens, job_templates, etc.
+    required: True
+  query_params:
+    description:
+      - The query parameters to search for in the form of key/value pairs.
+    type: dict
+    required: False
+    aliases: [query, data, filter, params]
+  expect_objects:
+    description:
+      - Error if the response does not contain either a detail view or a list view.
+    type: boolean
+    default: False
+    aliases: [expect_object]
+  expect_one:
+    description:
+      - Error if the response contains more than one object.
+    type: boolean
+    default: False
+  return_objects:
+    description:
+      - If a list view is returned, promote the list of results to the top-level of list returned.
+      - Allows using this lookup plugin to loop over objects without additional work.
+    type: boolean
+    default: True
+  return_all:
+    description:
+      - If the response is paginated, return all pages.
+    type: boolean
+    default: False
+  return_ids:
+    description:
+      - If response contains objects, promote the id key to the top-level entries in the list.
+      - Allows looking up a related object and passing it as a parameter to another module.
+      - This will convert the return to a string or list of strings depending on the number of selected items.
+    type: boolean
+    aliases: [return_id]
+    default: False
+  max_objects:
+    description:
+      - if C(return_all) is true, this is the maximum of number of objects to return from the list.
+      - If a list view returns more an max_objects an exception will be raised
+    type: integer
+    default: 1000
+
+notes:
+  - If the query is not filtered properly this can cause a performance impact.
+"""
+
+EXAMPLES = """
+- name: Load the UI settings
+  set_fact:
+    tower_settings: "{{ lookup('awx.awx.tower_api', 'settings/ui') }}"
+
+- name: Report the usernames of all users with admin privs
+  debug:
+    msg: "Admin users: {{ query('awx.awx.tower_api', 'users', query_params={ 'is_superuser': true }) | map(attribute='username') | join(', ') }}"
+
+- name: Lookup any users who are admins and get their objects directly
+  set_fact:
+    admin_user_object: "{{ lookup('awx.awx.tower_api', 'users', query_params={ 'is_superuser': true } ) }}"
+
+- name: Make sure user 'john' is an org admin of the default org if the user exists
```

Comment:
You mean make sure John is _not_ an admin of the default org, given that `state: absent`. Poor John.

(but speaking practically, it might be less awkward to delete the "state" and register" lines in the task)